File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> map;
        for(int num : nums){
            map[num]++;
        }
        int ans = -1;
        for(auto it : map){
            if(it.second == 1){
                ans = it.first;
                break;
            }
        }
        return ans;
    }
};

File: 0137-single-number-ii/0137-single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        int ans = 0;
        int n = nums.length;
        for(int i=0;i<=31;i++){
            int cnt = 0;
            for(int j=0;j<n;j++){
                if(((nums[j]>>i) & 1) == 1) cnt++;
            }
            if(cnt%3 != 0) ans = ans | (1<<i);
        }
        return ans;
    }
}